frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/initialize-vm.ts

```diff
@@ -20,35 +20,33 @@ import {
 } from '../../../wrapper/vm/cloud-init-data-helper';
 import { DataVolumeWrapper } from '../../../wrapper/vm/data-volume-wrapper';
 import { StorageUISource } from '../../../../components/modals/disk-modal/storage-ui-source';
-import { insertName, joinIDs } from '../../../../utils';
-import { VM_TEMPLATE_NAME_PARAMETER } from '../../../../constants/vm-templates';
+import { joinIDs } from '../../../../utils';
 import { CreateVMParams } from './types';
 import { isCustomFlavor } from '../../../../selectors/vm-like/flavor';
+import { DiskWrapper } from '../../../wrapper/vm/disk-wrapper';
+import { VM_TEMPLATE_NAME_PARAMETER } from '../../../../constants/vm-templates';
 
-const resolveDataVolumeName = (
-  dataVolumeWrapper: DataVolumeWrapper,
-  {
-    newName,
-    isTemplate,
-    isPlainDataVolume,
-  }: { newName: string; isTemplate: boolean; isPlainDataVolume: boolean },
-) => {
-  const finalName = insertName(
-    dataVolumeWrapper.getName(),
-    !isTemplate || isPlainDataVolume ? newName : VM_TEMPLATE_NAME_PARAMETER,
-  );
-  return dataVolumeWrapper.getType() === DataVolumeSourceType.PVC &&
-    finalName === dataVolumeWrapper.getPesistentVolumeClaimName()
-    ? joinIDs(finalName, 'clone')
-    : finalName;
+const resolveDataVolumeName = ({
+  diskName,
+  vmLikeEntityName,
+  isTemplate,
+  isPlainDataVolume,
+}: {
+  diskName: string;
+  vmLikeEntityName: string;
+  isTemplate: boolean;
+  isPlainDataVolume: boolean;
+}) => {
+  const baseName = joinIDs(vmLikeEntityName, diskName);
+  return isTemplate && !isPlainDataVolume ? baseName : joinIDs(baseName, getRandomChars(5));
```

Comment:
let's delegate this to `generateDataVolumeName` as well.

Reply:
Of course I missed the one I started with! haha. Fixed!